core/src/main/java/io/questdb/cutlass/pgwire/PGWireServer.java

```diff
@@ -81,6 +81,10 @@ public PGWireServer(
             workerPool.assign(i, new Job() {
                 private final IORequestProcessor<PGConnectionContext> processor = (operation, context) -> {
                     try {
+                        if (operation == IOOperation.HEARTBEAT) {
+                            context.getDispatcher().registerChannel(context, IOOperation.HEARTBEAT);
+                            return true;
```

Comment:
i feel heartbeat should be returning `false`  always

Reply:
fixed